### PR TITLE
Forgot to escape JSON string output

### PR DESCRIPTION
**Pull request by geeknoid**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4280)

### CODE REVIEW

File: src/Generators/Microsoft.Gen.ComplianceReports/Emitter.cs [hunk 1]
```diff
@@ -174,6 +174,10 @@ private void OutArray(string name, Action action)
 
     private void OutNameValue(string name, string value)
     {
+        value = value
```
Comment (sebastienros) [inline src/Generators/Microsoft.Gen.ComplianceReports/Emitter.cs:177]: Why not use `JsonSerializer.Serialize(value)` ?

Comment (xakep139) [reply src/Generators/Microsoft.Gen.ComplianceReports/Emitter.cs:177]: IIRC Generators can't reference packages (e.g. `System.Text.Json`). @geeknoid, is that correct?

Comment (sebastienros) [reply src/Generators/Microsoft.Gen.ComplianceReports/Emitter.cs:177]: And they target NS2.0, so that wouldn't be possible, ok.

**Review by sebastienros — APPROVED** [review]

File: src/Generators/Microsoft.Gen.ComplianceReports/Emitter.cs (hunk 1 above)
Comment (geeknoid) [reply src/Generators/Microsoft.Gen.ComplianceReports/Emitter.cs:177]: That's what I thought until a few minutes ago. Someone just sent me https://github.com/dotnet/roslyn/blob/main/docs/features/source-generators.cookbook.md#use-functionality-from-nuget-packages which shows the MSBuild magic to make this happen.

I'll create an issue to track upgrading to that approach.

**Pull request merged**